### ISSUE TITLE
aspect ratio incorrect

### ISSUE THREAD

**Issue by iscorporacion** [AmolGangadhare/flutter_barcode_scanner#109]
When I call the plugin it works perfect, but the aspect ratio I think is wrong. the object in my hand is a square and it looks like a rectangle

** Expected behavior **
should show a correct relationship

** Screenshots **
![Screenshot_20200730-041500](https://user-images.githubusercontent.com/15124507/88906235-cee12400-d21c-11ea-9f77-31527c29615a.jpg)

** Smartphone **
  - Device: Samsung A70
  - OS: Android 10

**Comment by MarceloaBatista** [comment]
I also have this problem.
Do you have any solution ?

**Comment by narritt** [comment]
Also have this problem. Reproduce in Samsung S8, Xiaomi Mi 8 Lite, OnePlus A3010

**Comment by iscorporacion** [comment]
> I also have this problem.
> Do you have any solution ?

I still can't solve it, it's not really a problem for me, but I'll be waiting to solve it

**Comment by marcomessa** [comment]
Same problem here, tried with Samsung Galaxy S8

**Comment by psrcek** [comment]
Same problem on Samsung A51 (SM-A515F)

**Comment by Himura2la** [comment]
Same problem, Xiaomi Mi A2

**Comment by Trend74X** [comment]
anyone found solution yet? i ran into this problem too

**Comment by ondine93** [comment]
Samsung A31 and One Plus 8 Pro With android 11 have the same issue, have you a workaround or a fix time?

**Comment by nicolasvahidzein** [comment]
I have this problem too, it didn't use to be like this, what happened?

**Comment by xritzx** [comment]
Yes, I am having the same issue with all of my android devices however it's working fine with ios 14.6 in iphone 11

**Comment by nicolasvahidzein** [comment]
Ok but we need this fixed on android. This is terrible.

**Comment by nicolasvahidzein** [comment]
Hello any news on this issue please?

**Comment by PatrickWulfe** [comment]
Also have this problem on both my Google Pixel 2 XL and Poco F1 on Pixel Experience.  Both are on Android 11.  It's also happening in my Pixel 5 Android emulator, which is also on 11.

**Comment by nicolasvahidzein** [comment]
Hello team. Any progress on this please? This is a big problem.

**Comment by nicolasvahidzein** [comment]
Hello guys. Any feedback on this? It has been quite a few months. Pretty please with a cherry on top.

**Comment by AndrewSheetMetal** [comment]
Same on Pixel 4a, I hoped that anyone had fixed this issue already :/

**Comment by stevenBang** [comment]
I have this problem too, 
Any news on this issue please?

**Comment by nicolasvahidzein** [comment]
This has been going on for way too long, i hope we can resolve this in 2022.

**Comment by stevenBang** [comment]
@nicolasvahidzein 
@iscorporacion 
I found the answer, I need to modify the android source code
**CameraSource.java**
Look for the selectSizePair method
```

    private static SizePair selectSizePair(Camera camera, int desiredWidth, int desiredHeight) {
        //steven 2022.06.02 mark
        //List<SizePair> validPreviewSizes = generateValidPreviewSizeList(camera);
        //
        // // The method for selecting the best size is to minimize the sum of the differences between
        // // the desired values and the actual values for width and height.  This is certainly not the
        // // only way to select the best size, but it provides a decent tradeoff between using the
        // // closest aspect ratio vs. using the closest pixel area.
        // SizePair selectedPair = null;
        // int minDiff = Integer.MAX_VALUE;
        // for (SizePair sizePair : validPreviewSizes) {
        //     Size size = sizePair.previewSize();
        //     int diff = Math.abs(size.getWidth() - desiredWidth) +
        //             Math.abs(size.getHeight() - desiredHeight);
        //     if (diff < minDiff) {
        //         selectedPair = sizePair;
        //         minDiff = diff;
        //     }
        // }
        //
        // return selectedPair;

        //steven 2022.06.02 add
        android.hardware.Camera.Size stevenSize = getOptimalPreviewSize(camera.getParameters().getSupportedPreviewSizes(), desiredWidth, desiredHeight);
        return (new SizePair(stevenSize, null));

    }

```

**Comment by nicolasvahidzein** [comment]
Wonderful news, can we implement this on our own or you will release an update?

If we should implement it can you give us detailed instructions on how to do it?

**Comment by ondine93** [comment]
It works, but you need to set getOptimalPreviewSize as static too.
I've forked the repo with the fix. If you wanna try replace in the pubspec.yaml of your project
`flutter_barcode_scanner: ^2.0.0`
with
```
  flutter_barcode_scanner: 
    git:
      url: https://github.com/ondine93/flutter_barcode_scanner.git
```

**Comment by nicolasvahidzein** [comment]
@ondine93 thank you so much. I will do that until the owner merges your PR.

Excellent work team.

**Comment by nicolasvahidzein** [comment]
Nope still not square aspect ratio @ondine93

**Comment by ondine93** [comment]
> Nope still not square aspect ratio @ondine93 

In which device are you testing?

**Comment by nicolasvahidzein** [comment]
@ondine93 Samsung A20, it should not matter should it? It's an Android.

**Comment by TheMeanCanEHdian** [comment]
~Confirmed that [ondine93](https://github.com/ondine93)'s changes worked in my testing.~

Edit: Initially I tested on a OnePlus 7T, I just tested on a Galaxy Fold 3, and it's actually still incorrect there (on both displays).

**Comment by nicolasvahidzein** [comment]
I thought i was crazy, ok thank God. So what is the next step? What could possibly break this?

**Comment by nicolasvahidzein** [comment]
Any luck in fixing it for all devices @ondine93 ?
